rust/auth-impls/src/lib.rs

```diff
@@ -0,0 +1,201 @@
+//! Hosts VSS protocol compliant [`Authorizer`] implementations.
+//!
+//! VSS is an open-source project designed to offer a server-side cloud storage solution specifically
+//! tailored for noncustodial Lightning supporting mobile wallets. Its primary objective is to
+//! simplify the development process for Lightning wallets by providing a secure means to store
+//! and manage the essential state required for Lightning Network (LN) operations.
+//!
+//! [`Authorizer`]: api::auth::Authorizer
+
+#![deny(rustdoc::broken_intra_doc_links)]
+#![deny(rustdoc::private_intra_doc_links)]
+#![deny(missing_docs)]
+
+use api::auth::{AuthResponse, Authorizer};
+use api::error::VssError;
+use async_trait::async_trait;
+use jsonwebtoken::{decode, Algorithm, DecodingKey, Validation};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// A JWT based authorizer, only allows requests with verified 'JsonWebToken' signed by the given
+/// issuer key.
+pub struct JWTAuthorizer {
+	jwt_issuer_key: DecodingKey,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct Claims {
+	sub: String,
+}
```

Comment:
Could you add docs or improve the naming here? It isn't clear to me what this is for.